### ISSUE TITLE
No error on wrong MXML value type

### ISSUE THREAD

**Issue by Harbs** [apache/royale-compiler#96]
Specifying a simple value for MXML which requires a complex value should cause a compile time error. One case which was just reported (https://stackoverflow.com/questions/58068839/how-to-use-graphic-svg-elements-with-apache-royale) is where an int was specified instead of an IFill for a Rect.

That causes the compiled MXML to have an empty array which cannot be resolved to a class at runtime. The runtime error is very opaque to users. A much better user experience would be an error "Expected IFill, but got int instead" with a pointer to the correct location in the MXML.

**Comment by joshtynjala** [comment]
Here is some Royale code that should result in a compiler error, but does not:

```mxml
<js:Application xmlns:fx="http://ns.adobe.com/mxml/2009"
	xmlns:js="library://ns.apache.org/royale/basic" xmlns:svg="library://ns.apache.org/royale/svg">
	<js:initialView>
		<js:View>
			<svg:Rect height="50" width="50" fill="0xff0000"/>
		</js:View>
	</js:initialView>
</js:Application>
```

A similar app created with Flex might look like this:

```mxml
<s:Application xmlns:fx="http://ns.adobe.com/mxml/2009"
	xmlns:s="library://ns.adobe.com/flex/spark">
	<s:Rect height="50" width="50" fill="0xff0000"/>
</s:Application>
```

Attempting to compile this code with the Flex SDK produces the following compiler error:

> Error: Initializer for 'fill': values of type mx.graphics.IFill cannot be represented in text.

**Comment by joshtynjala** [comment]
Closed in dc3b2cd139ae2982b7f767d6653e9cdf7b8256e9
